lib/kvdb/ikvdb.c

```diff
@@ -2448,6 +2445,13 @@ is_read_allowed(struct ikvs *kvs, struct hse_kvdb_txn *const txn)
     return txn && !kvs_txn_is_enabled(kvs) ? false : true;
 }
 
+static inline bool
+is_compression_allowed(struct kvdb_kvs *kk, const unsigned int flags)
+{
+    return (kk->kk_vcomp_default == VCOMP_DEFAULT_ON && !(flags & HSE_KVS_PUT_VCOMP_OFF)) ||
+        (kk->kk_vcomp_default == VCOMP_DEFAULT_OFF && (flags & HSE_KVS_PUT_VCOMP_ON));
+}
+
```

Comment:
Can you remove the (kk->kk_comp_default == VCOMP_DEFAULT_OFF &&) from the second line?